fbgemm_gpu/hip_kernel/split_tbe_fwd_hip.cpp

```diff
@@ -254,10 +253,13 @@ __device__ void split_tbe_forward_hip_kernel(
     if(bag_id >= batch)
         return ;
     int lane_id = threadIdx.x & (AMDGCN_WAVE_SIZE - 1);
-
+    int num_loads = (emb_dim + AMDGCN_WAVE_SIZE - 1) >> 6;
+    if( lane_id * num_loads > emb_dim)
```

Comment:
This logic is not enough. If we're <192 we're reading 3 columns per thread.  which means we'll have an uneven read at the end of a row.  This may require load_row_per_warp and others, to only do partial reads for threads at the end of a warp.  

Additional issue, 64, 128, and 256 use 1, 2, 4 reads per thread respectively...which we can easily sort out, but that 3 reads causes us to not be aligned.